### PR TITLE
A4A: Implement the WPCOM Plan selector in the new hosting page.

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR implements the WPCOM plan selector on the new hosting page.

Screenshot 2024-12-12 at 10 54 24 PM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1546

## Proposed Changes

* Implement a reusable Hosting plan section (later, we will reuse this as well for the Pressable plans).
* Create a new `A4AnumberInputV2` component derived from `A4AnumberInput`, incorporating a minor UI update that aligns with the latest design. We developed this separate component to prevent adding complexity to the `A4AnumberInput` code. Eventually, we will eliminate this as part of the project's post-cleanup process.
* Implement the new WPCOM plan section based on the new design.

## Why are these changes being made?

* This is part of the In-product hosting page v3 project.

## Testing Instructions

* Use the A4A live link and go to the `/marketplace/hosting/wpcom` page.
* Confirm you see the plan selector.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hero-section/style.scss [hunk 1]
```diff
@@ -99,11 +99,17 @@ $tab-background-selected: var(--color-surface);
 	}
 
 	&:first-child {
-		border-start-start-radius: 4px;
+		&,
+		.section-nav-tab__link {
+			border-start-start-radius: 4px;
+		}
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hero-section/style.scss:105]: While not the primary focus of this PR, it addresses a minor radius issue in the Header navigation menu.

File: client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hooks/use-compact-on-scroll.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ export default function useCompactOnScroll() {
 
 	const onTransitionEnd = useCallback( () => {
 		setIsTransitioning( false );
+		setLastScrollPosition( 0 );
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/hosting-overview-v3/hooks/use-compact-on-scroll.ts:16]: This is a small adjustment unrelated to this PR's main objective. There's a tendency for the logic to break and detect scroll-down after a transition where the elements shift upward. This modification guarantees that we reset the previous scroll position to 0 following the transition and ensure no miscalculation when determining a scroll up or down.

**Comment by cleacos** [comment]
I cannot see the selector:

image

The first time I loaded it, it appeared for a few seconds and then disappeared. I guess it is because I have more than 10 sites...

**Comment by jkguidaven** [comment]
> I cannot see the selector:
> 
> image
> The first time I loaded it, it appeared for a few seconds and then disappeared. I guess it is because I have more than 10 sites...

Yes, this was intentional. The selector is hidden if you have more than or equal to 10 WPCOM plans.

**Review by cleacos — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17065125

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17065125).

**Pull request merged**